ompi/mca/common/ompio/common_ompio_aggregators.c

```diff
@@ -1499,10 +1499,10 @@ static double cost_calc (int P, int P_a, size_t d_p, size_t b_c, int dim )
 	    
 	    n_ar = (float) P_y;
 	    if ( d_p > (P_a*b_c/P )) {
-		m_s = fmin(b_c / P_y, d_p);
+		m_s = fmin((float) b_c / (float)P_y, (float)d_p);
```

Comment:
`fmin` returns a double, perhaps `fminf` would be more appropriate.

Reply:
@gpaulsen thanks for the feedback. Taking your comment above also into consideration, I stayed with fmin, but converted all floats and doubles. Does this look ok for you?